app/javascript/src/App.jsx

```diff
@@ -1,10 +1,69 @@
-import React from 'react'
-import './App.scss'
+import React, { useEffect, useState } from "react";
+import { Route, Switch, BrowserRouter as Router } from "react-router-dom";
+import { either, isEmpty, isNil } from "ramda";
 
-export default function App() {
+import Signup from "./Authentication/Signup";
+import Login from "./Authentication/Login";
+import PrivateRoute from "./Common/PrivateRoute";
+import { getFromLocalStorage } from "./helpers/storage";
+import { setAuthHeaders, resetAuthTokens } from "./apis/axios";
+import PollForm from "./Poll/PollForm";
+import ShowPolls from "./Poll/ShowPolls";
+import Poll from "./Poll/Poll";
+import Header from "./Common/Header";
+import { clearLocalStorage } from "./helpers/storage";
+import authApi from "./apis/auth";
+
+const App = () => {
+  const [loading, setLoading] = useState(true);
+  const authToken = getFromLocalStorage("authToken");
+  const isLoggedIn = !either(isNil, isEmpty)(authToken) && authToken !== "null";
+
+  useEffect(() => {
+    // initializeLogger();
+    setAuthHeaders(setLoading);
```

Comment:
One piece of code should do only one thing. Don't combine setLoading logic along with setAuthHeaders logic.